Sources/Afluent/SequenceOperators/ThrottleSequence.swift

```diff
@@ -0,0 +1,198 @@
+                try Task.checkCancellation()
+                
+                let clock = self.clock
+                let interval = self.interval
+                let upstream = self.upstream
+                let state = self.state
+                
+                if iterator == nil {
+                    self.iterator = AsyncThrowingStream<(Element?, Element?), Error> { continuation in
+                        
+                        let intervalTask = DeferredTask {
+                            guard let intervalStartInstant = state.startInstant else { return }
+                            state.hasStartedInterval = true
+                            
+                            let intervalEndInstant = intervalStartInstant.advanced(by: interval)
+                            try await clock.sleep(until: intervalEndInstant, tolerance: nil)
+                            
+                            let firstElement = state.firstElement
+                            let latestElement = state.latestElement
+                            
+                            continuation.yield((firstElement, latestElement))
+                            
+                            state.firstElement = nil
+                            state.hasStartedInterval = false
+                        }
+                        
+                        let iterationTask = Task {
+                            do {
+                                for try await el in upstream {
+                                    if !state.hasSeenFirstElement {
+                                        continuation.yield((el, el))
+                                        state.hasSeenFirstElement = true
+                                        continue
+                                    }
+                                    if state.firstElement == nil {
+                                        state.startInstant = clock.now
+                                        state.firstElement = el
+                                        intervalTask.run()
+                                    }
+                                    state.latestElement = el
+                                }
+                                if state.hasStartedInterval {
+                                    continuation.yield((state.firstElement, state.latestElement))
+                                }
+                                continuation.finish()
+                            } catch {
+                                if state.hasStartedInterval {
+                                    continuation.yield((state.firstElement, state.latestElement))
+                                }
+                                continuation.finish(throwing: error)
+                            }
+                        }
+                        
+                        continuation.onTermination = { _ in
+                            intervalTask.cancel()
+                            iterationTask.cancel()
```

Comment:
This line (169) is untested.

Reply:
@aim2120 and I tried to figure out a way to test this that didn't involve leaking implementation details, but were unsuccessful.